crosspoint_reader/ws_client.py

```
@@ -153,18 +154,31 @@ def _recv_exact(self, n):
+            return []
+        else:
+            # Unix/Linux/Mac: select.select() works fine with sockets
+            while True:
+                r, _, _ = select.select([self.sock], [], [], 0)
+                if not r:
+                    break
+                opcode, payload = self._read_frame()
+                if opcode == 0x1:
+                    messages.append(payload.decode('utf-8', 'ignore'))
+                elif opcode == 0x8:
+                    raise WebSocketError('Connection closed')
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)